lib/index.js

```diff
@@ -33,14 +39,31 @@ function lint(editor) {
     processor = require('remark');
   }
 
+  if (recommendedWithoutConfig) {
```

Comment:
After doing this on another plugin that has many dependencies to check loading of, I would recommend moving the loading of these (and `engine` + `processor`) into a function, still with all the checks for already being loaded. Then call that function here and in the idle callback over in `activate`.

This way Atom will opportunistically load them during an idle period after activation, but still loads them immediately on first `lint()` call if they haven't already been loaded yet.